### ISSUE TITLE
DB/Professions: Goblin Mortar & The Mortar: Reloaded not available from Goblin Engineer Trainers

### ISSUE THREAD

**Issue by TQPS** [TrinityCore/TrinityCore#27642]
### Description

Both the Goblin Mortar and The Mortar: Reloaded recipes are not available on any of the Goblin Engineer trainers.

<details>

![image](https://user-images.githubusercontent.com/66294837/150214186-8d7f01ae-fbd5-4524-b60a-721128696ec2.png)

</details>

Addon used is Ackis Recipe List for WOTLK  

### Expected behaviour

Goblin Trainers (Tanaris, The Barrens & Dalaran) should offer these around Engineering level 205

(Although I have only tested classic content in game, a DB search shows they're missing altogether).

### Steps to reproduce the problem

1. Be a Goblin Engineer (Or have Goblin Engineer)
2. Visit a Goblin Engineer Trainer 
3. Recipes are missing

* Level shouldn't matter, it would just show up as unavailable if level requirement wasn't met

### Branch

3.3.5

### TC rev. hash/commit

TrinityCore rev. a650bd631a46 2022-01-16 18:50:28 +0100 (3.3.5 branch) (Win64, RelWithDebInfo, Static)

### Operating system

Windows 7 x64

### Custom changes

TrinityBots

**Comment by Aokromes** [comment]
maybe they where forgotten on the trainers rework.

**Comment by offl** [comment]
In case of https://tcubuntu.northeurope.cloudapp.azure.com/aowow/?npc=8126 I see no difference between old trainer (2011), current trainer and wowhead(2010)
There are only 3 goblin trainers and according to wowhead(2010) they all share same trainer spells
Maybe it's learned differently or trainer data is wrong everywhere

**Comment by offl** [comment]
_Originally posted by @anzz1 in https://github.com/TrinityCore/TrinityCore/issues/27739#issue-1125361075_

Affected trainers:  
Vazario Linkgrease ([8738](https://wotlkdb.com/?npc=8738)) Barrens
Didi The Wrench ([29513](https://wotlkdb.com/?npc=29513)) Dalaran
Nixx Sprocketspring ([8126](https://wotlkdb.com/?npc=8126)) Tanaris

```sql
-- Add Goblin Mortar and The Mortar: Reloaded recipes
DELETE FROM `trainer_spell` WHERE `TrainerId`=103 AND `SpellId` IN (13240,12716);
INSERT INTO `trainer_spell` (`TrainerId`, `SpellId`, `MoneyCost`, `ReqSkillLine`, `ReqSkillRank`, `ReqAbility1`, `ReqAbility2`, `ReqAbility3`, `ReqLevel`, `VerifiedBuild`) VALUES 
(103, 13240, 1500, 202, 205, 20222, 0, 0, 0, 0),
(103, 12716, 41000, 202, 205, 20222, 0, 0, 0, 0);
```

**Comment by offl** [comment]
@anzz1 MoneyCost is guessed here?

**Comment by offl** [comment]
Well, it works but gives me a feeling it looks a bit weird in trainer menu
Tried to find a sniff or screenshot of trainer menu but with no luck so far

**Comment by anzz1** [comment]
WH comment:
> [2007/03/15](https://www.wowhead.com/spell=13240/the-mortar-reloaded#comments:id=1560:reply=291129) (Patch 2.0.10)
> Bizarrely, this recipe will still yield skill points even if it has turned gray - right the way up to 375.

It might just be another of those cases where even Blizz themselves don't know what they're doing. :D 

But it seems that indeed using spell 12768 instead of having them both as recipes looks more right.
I used wowhead (tbc , classic, retail) as source for the skill req's and moneycost.

Ackis Recipe List is not a bad source either, especially if you use the very last version [v2.01-14-gfc58e9f](https://www.curseforge.com/wow/addons/arl/files/458020) released for 3.3.5 before cata, if you look at the changelog it seems that the dev really paid attention to detail to get everything right, but obviously there still can be errors and missing stuff.
